### DISCUSSION TITLE
iOS: app doesn't wake up for remote push notification

### DISCUSSION THREAD

**Discussion by ennabow** [invertase/react-native-firebase#6652] [Q&A]
A trouble: ios app doesn't show remote push notification in killed state.

The app does not wake up at 30 seconds in killed state.
Exception: the app wake up for remote push notification right after terminate app or for a several minutes after it. In other case app dosen't wake up.
Foreground, quit, background state - work correctly.

I saw similar issue - https://github.com/invertase/react-native-firebase/discussions/6408
I can't implement [PushKit](https://developer.apple.com/documentation/pushkit) because of I don't use voice messages or call. Someone suggested that this might cause problems when reviewing in AppStore.
Notification Service Extention is implemented.

Why remote push notification dont work in terminated state stable and doesnt wake up app?

signing & capabilities
```
+ background fetch 
+ remote notifications 
+ background processing
```

payload contains 'contentAvailable': true

project:
```
rect-native 0.64.2
react-native-firebase/app 16.4.0 (also tried 14.11.0)
react-native-firebase/messaging 16.4.0 (also tried 14.11.0)
notifee 7.0.4 (also tried 5.3.0)
```

firebase console contains
```
+ Authkey 
+ google-services.json 
```

devices:
iphone X OS 16.0.3
iphone 6s OS 13.3

Background App Refresh is ON

data-only notification looks like this (I'm using admin-sdk, NodeJS):
(I tried send mixed push with `notification` part but no difference)

```
messaging
            .send({
                token: fcmId,
                data: {
                    command: notification.command,
                    ...data
                },
                apns: {
                    payload: {
                        aps: {
                            contentAvailable: true,
                            mutableContent: true,
                            sound: 'default',
                        }
                    },
                    headers: {
                        'apns-push-type': 'background',
                        'apns-priority': '5',
                        'apns-topic': 'login.topnlab.ru',
                    },
                },
                android: {
                    priority: 'high',
                }
            })
```

I understand that firebase doesn't guarantee stable delivery of remote push notifications. But now app doesn't react at all on them.
It looks like I'm doing something wrong. Give me a hint, please, where could I be wrong?

**Comment by mikehardy** [comment]
I believe getting the device logs (you may see them via Console.app when using a real device plugged in to the computer) may help, especially with debugging enabled (similar to this, to enable: https://github.com/firebase/firebase-ios-sdk/issues/143#issue-243606481)

**Comment by aerdem4** [comment]
@ennabow were you able to solve this?
